packages/@react-stately/tooltip/src/useTooltipTriggerState.ts

```diff
@@ -10,23 +10,113 @@
  * governing permissions and limitations under the License.
  */
 
-import {TooltipManager} from './TooltipManager';
-import {useState} from 'react';
+import {TooltipTriggerProps} from '@react-types/tooltip';
+import {useEffect} from 'react';
+import {useId} from '@react-aria/utils';
+import {useOverlayTriggerState} from '@react-stately/overlays';
+
+export const TOOLTIP_DELAY = 1500; // this seems to be a 1.5 second delay, check with design
+export const TOOLTIP_COOLDOWN = 2000;
+
+interface TooltipTriggerStateProps extends TooltipTriggerProps {}
 
 export interface TooltipTriggerState {
-  open: boolean,
-  setOpen: (value: boolean) => void,
-  tooltipManager: TooltipManager
+  isOpen: boolean,
+  open: (immediate?: boolean) => void,
+  close: () => void
 }
 
-let tooltipManager = new TooltipManager();
+let tooltips = {};
+let globalWarmedUp = false;
+let globalWarmUpTimeout = null;
+let globalCooldownTimeout = null;
+
+
+export function useTooltipTriggerState(props: TooltipTriggerStateProps): TooltipTriggerState {
+  let {delay = TOOLTIP_DELAY} = props;
+  let {isOpen, open, close} = useOverlayTriggerState(props);
+  // this is a unique id for the tooltips in the map, it's not a dom id
+  let id = useId();
+
+  let ensureTooltipEntry = () => {
+    tooltips[id] = hideTooltip;
+  };
+
+  let closeOpenTooltips = () => {
+    for (let hideTooltipId in tooltips) {
+      if (hideTooltipId !== id) {
+        tooltips[hideTooltipId]();
```

Comment:
delete from map?

Reply:
thanks, this used to happen when they all had their own cooldowns, now that it's global, this is the right place to do that